release/models/bgp/openconfig-bgp-neighbor.yang

```diff
@@ -305,6 +312,20 @@ submodule openconfig-bgp-neighbor {
         due to an inbound connection request from a specified source prefix
         within a dynamic-neighbor-prefix.";
     }
+
+    leaf last-prefix-limit-exceeded {
+      type oc-types:timeticks64;
+      description
+        "This timestamp indicates the time that the BGP session last
+        violated a configured max prefix-limit for any AFI/SAFI combination
+        on the session.  The value is the timestamp in nanoseconds relative
+        to the Unix Epoch (Jan 1, 1970 00:00:00 UTC).
+
+        If the 'prefix-limit-exceeded' node is set to true for any AFI/SAFI
```

Comment:
Is this strictly true? I expect that if:

 * prefix-limit is tripped
 * connection-retry expires, and the local speaker sends an `OPEN`
 * for any reason the session is not re-established

then this leaf's value is not set to a new value (since we didn't hit max-prefix) and so the sentence is not correct.

It might be worth specifying when this value gets set to `nil`/deleted again -- should it be reset when a session is re-established? I'd imagine we don't necessarily need to say this because it'd be possible to tell whether a max-prefix event happened in the context of this session by looking at when the session was established.

Reply:
Taking the above workflow - lets assume we start from an established BGP session

* Now a prefix-limit is breached for any AFI/SAFI on the session - lets say IPv6 Unicast
* The `prefix-limit-exceeded` node for `neighbors/neighbor/afi-safis/afi-safi/ipv6-unicast/state` is set to `true` *AND* this leaf `last-prefix-limit-exceeded` has the breach time recorded
* connection retry now expires
* The _current status_ `prefix-limit-exceeded` leaf should be reset to false prior to attempting session re-establishment
* The `last-prefix-limit-exceeded` could still remain - This can always retain the last time it was tripped or can be completely reset at each new session establishment and is only ever set/updated during a breach event

If we have consensus on the above, I'll readjust the wording here to convey.  The `prefix-limit-exceeded` node is meant to convey the current state of the AFI/SAFI on the session which would only ever hold a true value for period of "last-prefix-limit-exceeded" + "restart-time" then flip back to false/reset during new session establishment.  

As far as resetting/deleting the `last-prefix-limit-exceeded` node vs. retaining a timestamp that could be many session establishments in the past (if sessions were not breached but rather reset for any other reason) - I'd be open to thoughts here.  On one hand, reset/delete holds less cross session state (which can be problematic for the implementation) but if both of these leafs are reset as soon as the retry timer expires, the history of the node state is lost (but should be gathered by other means of offline telemetry data/local logs, etc..)

Reply:
Regarding reset of last-prefix-limit-exceeded, it seems there is no clear right or wrong answer.  From an operational 
 point of view, it seems useful to only update last-prefix-limit-exceeded when an event occurs and NOT reset it.  This makes it similar to timestamps such as 'last-carrier-transition' for interfaces which is commonly treated this same way which also provides useful operational information.   Preserving this can help give insight into issues where there is momentary lapse of streaming telemetry.

Reply:
ack - addressed in https://github.com/openconfig/public/pull/532/commits/67c7c0e9948f96b6ff80ba521cd4f72b901a1a04